src/Cogworks.FindAndReplace/Web/API/FindAndReplaceApiController.cs

```diff
@@ -27,106 +25,58 @@ public FindAndReplaceApiController(IScopeProvider scopeProvider, IContentService
-                    {
-                        VarcharValue = x.VarcharValue,
-                        TextValue = x.TextValue,
-                        VersionId = x.Id,
-                        NodeName = x.Text,
-                        PropertyAlias = x.Alias,
-                        PropertyName = x.Name,
-                    });
-
-                return result.ToList();
+                var sqlParams = new
+                {
+                    phrase = $"%{phrase}%",
+                    contentId = $"%{contentId}%"
+                };
+
+                var sqlQuery = $@"
+                    SELECT
+                          [upd].[varcharValue] AS [VarcharValue]
+                        , [upd].[textValue] AS [TextValue]
+                        , [cv].[id] AS [VersionId]
+                        , [pcv].[text] AS [NodeName]
+                        , [upt].[Alias] AS [PropertyAlias]
+                        , [upt].[Name] AS [PropertyName]
+                    FROM [umbracoDocument]
+                        INNER JOIN [umbracoContent] ON ([umbracoContent].[nodeId] = [umbracoDocument].[nodeId])
+                        INNER JOIN [umbracoNode] ON ([umbracoNode].[id] = [umbracoContent].[nodeId])
+                        INNER JOIN [umbracoContentVersion] [cv] ON ([cv].[nodeId] = [umbracoDocument].[nodeId])
+                        INNER JOIN [umbracoDocumentVersion] ON ([umbracoDocumentVersion].[id] = [cv].[id])
+                        LEFT JOIN [umbracoContentVersion] [pcv]
+                            INNER JOIN [umbracoDocumentVersion] [pdv] ON (([pdv].[id] = [pcv].[id]) AND [pdv].[published] = 1)
+                        ON ([pcv].[nodeId] = [umbracoDocument].[nodeId])
+                        LEFT JOIN [umbracoContentVersionCultureVariation] [ccv]
+                            INNER JOIN [umbracoLanguage] [lang] ON (([lang].[id] = [ccv].[languageId]) AND ([lang].[languageISOCode] =N'[[[ISOCODE]]]'))
+                        ON ([cv].[id] = [ccv].[versionId])
+                        INNER JOIN [umbracoPropertyData] [upd] ON ([upd].[versionId] = [cv].[id])
+                        INNER JOIN [cmsPropertyType] [upt] ON ([upt].[id] = [upd].[propertyTypeId])
+                    WHERE ([cv].[current] = 1)
+                        AND ([umbracoDocument].[published] = 1)
+                        AND ([umbracoNode].[path] LIKE upper(@contentId))
+                        AND
+                        (
+                            (
+                                [upd].[textValue] is not null AND [upd].[textValue] LIKE @phrase
+                            )
+                            OR
+                            (
+                                [upd].[varcharValue] is not null AND [upd].[varcharValue] LIKE @phrase
+                            )
+                        )
+                    ORDER BY ([umbracoDocument].[nodeId])";
+
+                var sql = scope.SqlContext.Sql(sqlQuery, sqlParams);
```

Comment:
Also secure for SQL Injection through SQL Parameter (NPoco inside mechanism)

- https://stackoverflow.com/questions/44907263/does-petapocos-querystring-query-parameters-method-protect-from-sql-injectio
- https://github.com/schotime/NPoco/blob/d707af2bf59f5c70c1d147bc6974d3cdd52bb05b/src/NPoco/ParameterHelper.cs#L16

Reply:
Exactly what needs to be done with it. Great work @aochmann !